pkg/config/parameter/file/file.go

```diff
@@ -33,34 +34,60 @@ var FileParameterSerde = parameter.ParameterSerDe{
 }
 
 type FileParameter struct {
-	WorkingDir string
-	Folder     string
-	Path       string
+	WorkingDir           string
+	Folder               string
+	Path                 string
+	referencedParameters []parameter.ParameterReference
 }
 
 func (f *FileParameter) GetType() string {
 	return FileParameterType
 }
 
 func (f *FileParameter) GetReferences() []parameter.ParameterReference {
-	// file parameters cannot have references
-	return []parameter.ParameterReference{}
+	if f.referencedParameters == nil {
```

Comment:
nit: is that check/initialized value return necessary? shouldn't a nil slice generally have the same effect as an empty initialized reference slice when used?

Reply:
depends on the usage, a nil slice is nil wheras the other one is not, but yes, for example `len` of both schould be 0.

Reply:
do other implemetations of GetReferences hold/make that promise that a zero lenght slice is returned instead of nil value?